main.js

```diff
@@ -0,0 +1,128 @@
+const core = require('@actions/core');
+const fs = require('fs');
+const path = require('path');
+const axios = require('axios').default;
+
+const srcPath = core.getInput('srcPath', { required: true });
+const dstPath = core.getInput('dstPath', { required: true })
+const dropboxToken = core.getInput('token', { required: true });
+core.setSecret(dropboxToken);
+
+// 150MB per upload limit on the Dropbox API
+const MAX_UPLOAD_BYTES = 157286400;
+var filesToUpload = [];
+
+checkDropboxAuthentication();
+
+function checkDropboxAuthentication() {
+  const url = "https://api.dropboxapi.com/2/check/user";
+  const data = {
+    query: "Check Authentication"
+  }
+
+  axios({
+    url: url,
+    method: 'post',
+    headers: {
+      'Authorization' : 'Bearer ' + dropboxToken,
+      'Content-Type' : 'application/json'
+    },
+    data : JSON.stringify(data)
+  }).then(function (response) {
+    onAuthenticationSuccess(response);
+  }).catch(function (error) {
+    onAuthenticationFail(error);
+  });
+}
+
+function onAuthenticationSuccess(response) {
+  console.log('Auth response: ' + response.status);
+  console.log('Auth response: ' + response.statusText);
+
+  getDirFilesRecursive(srcPath);
+  uploadBuild();
+}
+
+function onAuthenticationFail(error) {
+  console.log(error);
+  core.setFailed(error);
+}
+
+function getDirFilesRecursive(dir) {
+  fs.readdirSync(dir).forEach(item => {
+    const fullPath = path.join(dir, item);
+    if (fs.lstatSync(fullPath).isDirectory()) {
+      getDirFilesRecursive(fullPath);
+    } else {
+      filesToUpload.push(fullPath);
+    }
+  });
+}
+
+function uploadBuild() {
+  uploadFile(filesToUpload[0], onUploadSuccess, onUploadFail);
+}
+
+function uploadFile(filePath, onSuccess, onFail) {
+  console.log("File: " + filePath);
+  const fileDstPath = filePath.replace(srcPath, dstPath);
+  console.log("Uploading to: " + fileDstPath);
+
+  const fileContent = fs.readFileSync(filePath);
+  const apiArgs = {
+    path: fileDstPath,
+    mute: true
+  }
+
+  const url = "https://content.dropboxapi.com/2/files/upload";
+  axios({
+    url: url,
+    method: 'post',
+    maxContentLength: MAX_UPLOAD_BYTES,
+    headers: {
+      'Authorization' : 'Bearer ' + dropboxToken,
+      'Content-Type' : 'application/octet-stream',
+      'Dropbox-API-Arg' : JSON.stringify(apiArgs)
+    },
+    data : fileContent
+  }).then(function (response) {
+    onSuccess(filePath, response);
+  }).catch(function (error) {
+    onFail(error);
+  });
+}
+
+function onUploadSuccess(localFilePath, response) {
+  console.log('Upload Successful: ' + localFilePath + '\n');
+
+  // As we only attempt to upload the next file on success of the previous, index should always be 0
+  let index = filesToUpload.indexOf(localFilePath);
```

Comment:
What order are the files uploaded in? If always first why is the search needed? Apologies if I'm being thick.

Reply:
Bit of a hangover from old code which was attempting to upload asynchronously but this gave me a tonne of rate limit errors.  Will remove to avoid confusion